app/src/main/java/com/vyvu/musikapp/repo/Mp3Repository.kt

```diff
@@ -0,0 +1,27 @@
+package com.vyvu.musikapp.repo
+
+import android.app.DownloadManager
+import android.content.ContentResolver
+import com.vyvu.musikapp.contract.modelcontract.Mp3Contract
+import com.vyvu.musikapp.model.Mp3
+
+
+class Mp3Repository(private val mp3Contract: Mp3Contract) : Mp3Contract {
+    init { instance = this }
```

Comment:
instance khởi tạo lại mỗi khi init thì đâu gọi là instance nhỉ ?

Reply:
t chỉ gọi hàm khởi tạo đúng 1 lần trong DataManager, lúc dùng sẽ là Mp3Repository.instance, không biết bị chạy lại không nữa, để t test lại xem sao

Reply:
t test rồi nhé, không gọi lại
t thử đặt Log ở trong khối init, kết quả là chỉ in ra 1 lần khi khởi tạo trong DataManager, còn khi gọi Mp3Repository.instance thì không có output nữa

Reply:
cách triển khai instance này ổn nhưng phải dùng lateinit để tránh khai báo instance bằng null, mà t thấy mọi người không thích lateinit cho lắm nên t cũng hơi ngại 😅